include/openssl/ec_key.h

```diff
@@ -356,6 +328,82 @@ OPENSSL_EXPORT EC_KEY *o2i_ECPublicKey(EC_KEY **out_key, const uint8_t **inp,
 OPENSSL_EXPORT int i2o_ECPublicKey(const EC_KEY *key, unsigned char **outp);
 
 
+// EC_KEY_METHOD
+
+// ECDSA_FLAG_OPAQUE specifies that this EC_KEY_METHOD does not expose its key
+// material. This may be set if, for instance, it is wrapping some other crypto
+// API, like a platform key store. Use |EC_KEY_METHOD_set_flag| to set
+// this flag on an |EC_KEY_METHOD|.
+// This was supported in ECDSA_METHOD previously and is not set by default.
+#define ECDSA_FLAG_OPAQUE 1
+
+// EC_KEY_get_default_method returns a reference to the default
+// |EC_KEY| implementation. The returned |EC_KEY_METHOD| object is
+// zero-initialized. This is different from OpenSSL which returns function
+// pointers to the default implementations within the |EC_KEY_METHOD| struct.
+// We do not do this to make it easier for the compiler/linker to drop unused
+// functions. The wrapper functions will select the appropriate
+// |ec_key_default_*| implementation.
+OPENSSL_EXPORT const EC_KEY_METHOD *EC_KEY_get_default_method(void);
+
+// EC_KEY_OpenSSL calls |EC_KEY_get_default_method|.
+OPENSSL_EXPORT const EC_KEY_METHOD *EC_KEY_OpenSSL(void);
+
+// EC_KEY_METHOD_new returns a newly allocated |EC_KEY_METHOD| object. If the
+// input parameter |eckey_meth| is non-NULL, the function pointers within the
+// returned |EC_KEY_METHOD| object will be initialized to the values from
+// |eckey_meth|. If |eckey_meth| is NULL, the returned object will be
+// zero-initialized.
+OPENSSL_EXPORT EC_KEY_METHOD *EC_KEY_METHOD_new(const EC_KEY_METHOD *eckey_meth);
+
+// EC_KEY_METHOD_free frees the memory associated with |eckey_meth|
+OPENSSL_EXPORT void EC_KEY_METHOD_free(EC_KEY_METHOD *eckey_meth);
+
+// EC_KEY_set_method sets |meth| on |ec|. Currently, AWS-LC does not support
+// setting the compute_key, copy, keygen, set_group, set_private, set_public,
+// verify, verify_sig fields. These must be NULL or the function will fail.
+// Returns zero on failure and one on success.
+OPENSSL_EXPORT int EC_KEY_set_method(EC_KEY *ec, const EC_KEY_METHOD *meth);
```

Comment:
Since we don't currently provide setters for these operations, it wouldn't be possible to set those fields. I can change the documentation to be clearer about this - I think leaving the check in there should be okay just to make sure those fields are NULL until we support them.